go/vt/vtgate/planbuilder/testdata/select_cases_with_default.json

```diff
@@ -61,9 +61,9 @@
         "Vindex": "user_index"
       },
       "TablesUsed": [
-        "second_user.dual",
+        "user.dual",
```

Comment:
This seems OK to me, maybe even an improvement. LMK if we prefer the current plan over this.